File: host/src/model/clist/cltypes.py

deck = 0
hand = 1
active = 2
grave = 3
special = 4
other = 5

names = {}
names[deck]='deck'
names[hand]='hand'
names[active]='active'
names[grave]='grave'
names[special]='special'
names[other]='other'

size = len(names)+1

full = range(0,size)

class Visibility:
	def __init__(self):
		self.visible = range(0,size)
		self.visible[deck] = (False,False)
		self.visible[hand] = (True,False)
		self.visible[active] = (True,True)
		self.visible[grave] = (True,True)
		self.visible[special] = (False,False)
		self.visible[other] = (True,False)

File: src/ppbackend/model/shared/CLTypes.java
package ppbackend.model.shared;

import java.util.HashMap;

public class CLTypes{
	public static int deck = 0;
	public static int hand = 1;
	public static int active = 2;
	public static int grave = 3;
	public static int special = 4;
	public static int other = 5;

	public static int size = 6;

	public static String[] names = {
		"deck",
		"hand",
		"active",
		"grave",
		"special",
		"other"
	};

	public static HashMap<String,Integer> values = new HashMap<String,Integer>();
	static{
		values.put("deck",deck);
		values.put("hand",hand);
		values.put("active",active);
		values.put("grave",grave);
		values.put("special",special);
		values.put("other",other);
	}
}
